# Data Analysis



In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

amazon = pd.read_csv("https://raw.githubusercontent.com/amanzanares410/ds4002-project1/refs/heads/main/DATA/amazon_clothing_reviews.csv")
print(amazon.shape)
amazon.head()


(6525, 11)


Unnamed: 0  Clothing ID  Age                    Title  \
0           0          767   33                      NaN   
1           1         1080   34                      NaN   
2           2         1077   60  Some major design flaws   
3           3         1049   50         My favorite buy!   
4           4          847   47         Flattering shirt   

                                         Review Text  Rating  Recommended IND  \
0  Absolutely wonderful - silky and sexy and comf...       4                1   
1  Love this dress!  it's sooo pretty.  i happene...       5                1   
2  I had such high hopes for this dress and reall...       3                0   
3  I love, love, love this jumpsuit. it's fun, fl...       5                1   
4  This shirt is very flattering to all due to th...       5                1   

   Positive Feedback Count   Division Name Department Name Class Name  
0                        0       Initmates        Intimate  Intimates  
1                        4         General         Dresses    Dresses  
2                        0         General         Dresses    Dresses  
3                        0  General Petite         Bottoms      Pants  
4                        6         General            Tops    Blouses

The data dimensions are 6525 x 11, so there are 6525 observations of 11
variables.



In [2]:
fashion_nova = pd.read_csv('https://raw.githubusercontent.com/amanzanares410/DS4002-Project1/refs/heads/main/DATA/fashionnova_reviews.csv')
print(fashion_nova.shape)
fashion_nova.head()

(131980, 9)


Reviewer Name                     Profile Link Country Review Count  \
0        Champagne  /users/66c78240087b6269ffbcb5fb      US     1 review   
1      Vg customer  /users/6618fdb53d4198001210cbe7      VG    3 reviews   
2  Colleen Burgher  /users/64e9595206be1a001244ff73      US    3 reviews   
3            R.G.M  /users/66c58ad1c6ab36352a08f57a      US     1 review   
4   Rosalyn Cousar  /users/60ad4b6ef3788e001adbb8e3      US    5 reviews   

                Review Date                  Rating  \
0  2024-08-22T20:24:02.000Z  Rated 5 out of 5 stars   
1  2024-08-21T05:43:11.000Z  Rated 5 out of 5 stars   
2  2024-08-21T17:09:14.000Z  Rated 5 out of 5 stars   
3  2024-08-21T08:36:03.000Z  Rated 5 out of 5 stars   
4  2024-08-22T00:46:16.000Z  Rated 3 out of 5 stars   

                          Review Title  \
0    I love ordering from fashion nova   
1   Top tier content for fashion nova    
2  Prices and quality of products are…   
3               Great customer service   
4                    False advertising   

                                         Review Text Date of Experience  
0  I love ordering from fashion nova. The clothes...    August 22, 2024  
1  Always amazing clothes and the fast shipping i...    August 18, 2024  
2  Prices and quality of products are GREAT Would...    August 21, 2024  
3  Great customer service. I was helped until the...    August 20, 2024  
4  Disappointing experience. You don’t live up to...    August 21, 2024

In [3]:
print(sum(amazon['Review Text'].isna()))
print(sum(amazon['Title'].isna()))
print(sum(amazon['Rating'].isna()))

print(sum(fashion_nova['Review Text'].isna()))
print(sum(fashion_nova['Review Title'].isna()))
print(sum(fashion_nova['Rating'].isna()))

229
1068
0
0
48
0


In [4]:
amazon = amazon.dropna(subset=['Review Text'])

rating_column = fashion_nova['Rating'].str.extract(r'Rated (\d) out of 5 stars').astype(int)
fashion_nova['Rating'] = rating_column

In [7]:
print(sum(amazon['Review Text'].isna()))
print(sum(fashion_nova['Review Text'].isna()))

0
0


In [5]:
df = pd.DataFrame()
df['Fashion Nova_shipping'] = fashion_nova['Review Text'].str.count(r'\bshipping\b')
df['Amazon_shipping'] = amazon['Review Text'].str.count(r'\bshipping\b')
df['Fashion Nova_quality'] = fashion_nova['Review Text'].str.count(r'\bquality\b')
df['Amazon_quality'] = amazon['Review Text'].str.count(r'\bquality\b')

In [6]:
# prompt: print count of each column in dataframe df

for column in df.columns:
  print(f"Counts for {column}:")
  print(df[column].value_counts())
  print("\n")


Counts for Fashion Nova_shipping:
Fashion Nova_shipping
0     124276
1       6466
2        921
3        215
4         56
5         27
6         10
7          5
8          2
10         1
9          1
Name: count, dtype: int64


Counts for Amazon_shipping:
Amazon_shipping
0.0    6282
1.0      14
Name: count, dtype: int64


Counts for Fashion Nova_quality:
Fashion Nova_quality
0    123654
1      8071
2       222
3        27
4         4
6         1
5         1
Name: count, dtype: int64


Counts for Amazon_quality:
Amazon_quality
0.0    5707
1.0     564
2.0      23
3.0       2
Name: count, dtype: int64


